src/runtime/Makefile

```diff
@@ -168,7 +168,21 @@ DEFAULTEXPFEATURES := []
 DEFENTROPYSOURCE := /dev/urandom
 
 DEFDISABLEBLOCK := false
-DEFSHAREDFS_QEMU_VIRTIOFS := virtio-fs
+DEFSHAREDFS := virtio-fs
+# virtio-fs support is relatively recent on s390x
+ifeq ($(ARCH),s390x)
+    ifeq (,$(shell \
+            qmp_sock=/tmp/qemu-capabilities.monitor.sock; \
+            pidfile=/tmp/qemu-capabilities.pidfile; \
+            $(QEMUPATH) -S -no-user-config -nodefaults -nographic -M none -daemonize \
+                        -qmp unix:$$qmp_sock,server,nowait -pidfile $$pidfile; \
+            echo '{"execute": "qmp_capabilities"} {"execute": "qom-list-types"}' | socat - UNIX:$$qmp_sock | \
+                tail -1 | jq ".return | .[].name" | grep -F vhost-user-fs-ccw; \
+            kill $$(cat $$pidfile); \
+            rm -f $$qmp_sock $$pidfile))
+        DEFSHAREDFS := virtio-9p
+    endif
```

Comment:
@Jakob-Naucke, please, bear with me here and let me see if I properly understand what you're doing here.

You define virtio-fs a the default fs, then you check, for s390x, whether the QEMU on the machine used for building kata-containers, has support to virtio-fs and, if not, you fallback to virtio-9p.

Do I get it correctly?  If so, what ensures that the machine where kata-containers will be running will have a QEMU as new as the one present in the machine where it was built?

I see this as something a distro should ensure that will work, or having an "on the fly" fallback mechanism as part of the runtime code.  But not something that can be ensure in at the build time.

WDYT?

Reply:
> You define virtio-fs a the default fs, then you check, for s390x, whether the QEMU on the machine used for building kata-containers, has support to virtio-fs and, if not, you fallback to virtio-9p.
> 
> Do I get it correctly?

Yes, that is what I'm doing here.

>  If so, what ensures that the machine where kata-containers will be running will have a QEMU as new as the one present in the machine where it was built?

Indeed, I had not considered the possibility of having a build machine with a QEMU setup different from the target machine.

> I see this as something a distro should ensure that will work, or having an "on the fly" fallback mechanism as part of the runtime code. But not as something that can be ensured at the build time.
> 
> WDYT?

I'd be against falling back to 9p in a way that isn't very clear to the user when virtio-fs is configured, i.e. a journal entry wouldn't be enough.
I don't know whether there is a surefire way to do that except for fatal messages, so maybe I'd keep it a fatal message.
Then I'd probably want to replace what would happen now when you'd run virtio-fs without support in QEMU (`'vhost-user-fs-ccw' is not a valid device model name`, see above) with a more understandable message about virtio-fs not being available.

Does this sound reasonable?

Reply:
That sounds super reasonable, and I also have the preference for not silently falling back to 9p.
A huge fatal message, and a better error report is, indeed, the way to go.

Thanks, @Jakob-Naucke!